src/common/abstractPromote.ts

```diff
@@ -77,18 +79,25 @@ export abstract class PromoteCommand<T extends typeof SfCommand> extends SfComma
     this.sourceStageId = this.getSourceStageId();
     const asyncOperationId: string = await this.requestPromotion(doceOrg);
 
-    // TODO: move this to logger service
-    this.log(`Job ID: ${asyncOperationId}`);
-    await DeployPipelineCache.set(asyncOperationId, {});
+    const outputService: OutputService = OutputServiceFactory.forDeployment(doceOrg.getConnection());
```

Comment:
I would use constructor injection for this (and the resume command).  It will make it easier to test.